### ISSUE TITLE
Nvda is not reading the site suggestions in the address field in Microsoft Edge

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#7554]
When opening microsoft hedge the focus goes to a search box where we can enter the site to be opened.
When typing the site as nv access, nvda makes the opening sound of the suggestions, however the suggestions are not read.
This works correctly in the start menu.
### Steps to reproduce:
Open Microsoft hedge.
Type in the site to open as an example type
www.nvd
Note that the hint sound is executed, but note that not a hint is read by the synthesizer.
Expected suggestions sound is performed, and the synthesizer reads the displayed suggestions as you type
www.nvd
The synthesizer should read at some point
www.nvda-project.org
When searching with the right and left arrows you can see that the hedge has filled correctly with the suggested site.
### Expected behavior:
> Tell us what should happen.

### Actual behavior:
Only the hint sound is played, but the synthesizer does not read the suggestions

### System configuration:
NVDA version:
next 14373

NVDA Installed or portable:
instaled

Windows version:
windows 10 15063.540 64bits
Name and version of other software in use when reproducing the issue:
microsoft hedge

### Other questions:

Does the issue still occur after restarting your PC?
yes

Have you tried any other versions of NVDA?
no

**Comment by fernando-jose-silva** [comment]
IO - inputCore.InputManager.executeGesture (13:45:26.855):
Input: kb(laptop):enter
IO - speech.speak (13:45:27.134):
Speaking [u'Microsoft Edge']
DEBUGWARNING - RPC process 14184 (MicrosoftEdge.exe) (13:45:27.615):
Thread 10400, build\x86_64\remote\ia2Support.cpp, IA2Support_inProcess_initialize, 121:
disabling IA2 support

IO - speech.speak (13:45:27.994):
Speaking [u'Microsoft Edge  janela']
IO - speech.speak (13:45:28.085):
Speaking [u'Atualizar  bot\xe3o']
IO - speech.speak (13:45:28.361):
Speaking [u'Pesquisar ou inserir endere\xe7o Web  edi\xe7\xe3o  recolhido']
IO - speech.speak (13:45:28.374):
Speaking [u'em branco']
DEBUGWARNING - RPC process 10980 (MicrosoftEdgeCP.exe) (13:45:30.322):
Thread 13744, build\x86_64\remote\ia2Support.cpp, IA2Support_inProcess_initialize, 121:
disabling IA2 support

IO - inputCore.InputManager.executeGesture (13:45:36.107):
Input: kb(laptop):w
IO - speech._speakSpellingGen (13:45:36.131):
Speaking character u'W'
IO - inputCore.InputManager.executeGesture (13:45:40.003):
Input: kb(laptop):w
IO - speech._speakSpellingGen (13:45:40.023):
Speaking character u'W'
IO - inputCore.InputManager.executeGesture (13:45:41.713):
Input: kb(laptop):w
IO - speech._speakSpellingGen (13:45:41.742):
Speaking character u'W'
IO - inputCore.InputManager.executeGesture (13:45:43.075):
Input: kb(laptop):.
IO - speech.speakTypedCharacters (13:45:43.118):
typed word: WWW
IO - speech.speak (13:45:43.118):
Speaking [u'WWW']
IO - speech._speakSpellingGen (13:45:43.148):
Speaking character u'ponto'
IO - inputCore.InputManager.executeGesture (13:45:46.252):
Input: kb(laptop):n
IO - speech._speakSpellingGen (13:45:46.279):
Speaking character u'N'
IO - inputCore.InputManager.executeGesture (13:45:48.161):
Input: kb(laptop):v
IO - speech._speakSpellingGen (13:45:48.184):
Speaking character u'V'
IO - inputCore.InputManager.executeGesture (13:45:50.188):
Input: kb(laptop):d
IO - speech._speakSpellingGen (13:45:50.233):
Speaking character u'D'
IO - inputCore.InputManager.executeGesture (13:45:52.382):
Input: kb(laptop):rightArrow
IO - speech.speak (13:45:52.437):
Speaking [u'em branco']
IO - inputCore.InputManager.executeGesture (13:45:54.575):
Input: kb(laptop):leftArrow
IO - speech._speakSpellingGen (13:45:54.642):
Speaking character u'barra'
IO - inputCore.InputManager.executeGesture (13:45:55.864):
Input: kb(laptop):leftArrow
IO - speech._speakSpellingGen (13:45:55.930):
Speaking character u'g'
IO - inputCore.InputManager.executeGesture (13:45:56.878):
Input: kb(laptop):leftArrow
IO - speech._speakSpellingGen (13:45:56.937):
Speaking character u'r'
IO - inputCore.InputManager.executeGesture (13:46:03.259):
Input: kb(laptop):NVDA+f1
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (13:46:03.530):
Developer info for navigator object:
name: u'Pesquisar ou inserir endere\xe7o Web'
role: ROLE_EDITABLETEXT
states: STATE_FOCUSABLE, STATE_EXPANDED, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionSearchFieldUIA object at 0x0AA79B70>
Python class mro: (<class 'NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionSearchFieldUIA'>, <class 'NVDAObjects.behaviors.EditableTextWithoutAutoSelectDetection'>, <class 'editableText.EditableTextWithoutAutoSelectDetection'>, <class 'NVDAObjects.behaviors.EditableText'>, <class 'editableText.EditableText'>, <class 'NVDAObjects.UIA.SearchField'>, <class 'NVDAObjects.behaviors.EditableTextWithSuggestions'>, <class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: ''
location: (250, 74, 846, 32)
value: u'WWW.NVDa-project.org/'
appModule: <'microsoftedge' (appName u'microsoftedge', process ID 14184) at address aa971b0>
appModule.productName: u'Microsoft Edge'
appModule.productVersion: u'11.00.15063.447'
TextInfo: <class 'NVDAObjects.UIA.UIATextInfo'>
windowHandle: 263104
windowClassName: u'Windows.UI.Core.CoreWindow'
windowControlID: 0
windowStyle: 1409286144
windowThreadID: 13572
windowText: u'Microsoft Edge'
displayText: u''
UIAElement: <POINTER(IUIAutomationElement) ptr=0x66aab00 at a9ebbc0>
UIA automationID: addressEditBox
UIA frameworkID: XAML
UIA runtimeID: (42, 263104, 3, 1)
UIA providerDescription: [pid:14184,providerId:0x0 Main(parent link):Unidentified Provider (unmanaged:Windows.UI.Xaml.dll)]
UIA className: RichEditBox

**Comment by josephsl** [comment]
Hi,

I once tried to come up with a fix for it to no avail, as Edge's address omnibar uses an odd implementation of controller for event. I think this was brought up before (@JCSTeh, do you remember our discussion of value change events and Edge?).

Thanks.

**Comment by fernando-jose-silva** [comment]
Thank you so much for looking at this.

 

De: Joseph Lee [mailto:notifications@github.com] 
Enviada em: sexta-feira, 1 de setembro de 2017 14:07
Para: nvaccess/nvda <nvda@noreply.github.com>
Cc: fernando-jose-silva <fernandoj.silva21@gmail.com>; Author <author@noreply.github.com>
Assunto: Re: [nvaccess/nvda] Nvda is not reading the site suggestions in the address field in Microsoft Edge (#7554)

 

Hi,

I once tried to come up with a fix for it to no avail, as Edge's address omnibar uses an odd implementation of controller for event. I think this was brought up before (@jcsteh <https://github.com/jcsteh> , do you remember our discussion of value change events and Edge?).

Thanks.

**Comment by Adriani90** [comment]
This works as expected when using the addon Windows 10 app essentials but it is not implemented in NVDA's core.
I am not sure if and for how long the current native edge version will be supported now that Edge chromium came out.

**Comment by josephsl** [comment]
Hi,

Update: CC @LeonardDer, as his UIA auto select work will resolve this (finally).

Thanks.
